spec/models/tag_spec.rb

```diff
@@ -32,13 +32,62 @@
     it "sorts the tags in trending order" do
       7.times { create(:tag, name: "Kaizen") }
       expect(Tag.get_tags_that_are(arg)).to be_a Hash
-      expect(Tag.get_tags_that_are(arg).first.first).to eq("Kaizen")
+      expect(Tag.get_tags_that_are(arg).first.first).to eq("kaizen")
       expect(Tag.get_tags_that_are(arg).first.last).to eq(9)
-      expect(Tag.get_tags_that_are(arg).keys.last).to eq("Amity")
+      expect(Tag.get_tags_that_are(arg).keys.last).to eq("amity")
       expect(Tag.get_tags_that_are(arg).values.last).to eq(3)
      end
   end
 
+  describe ".process_tags" do
+    it "returns an array of new instances of tags if they don't exist" do
+      tags = "tag1,tag2,tag3"
+      result = Tag.process_tags(tags)
+      expect(result.length).to be 3
+      expect(result.map(&:name)).to eql ['tag1', 'tag2', 'tag3']
+      expect(result.map(&:new_record?)).to eql [true, true, true]
+      expect(result.first).to be_a Tag
+    end
+
+    it "returns an array of the objects if they already exist" do
+      tags = []
+      3.times { tags << create(:tag).name }
+      result = Tag.process_tags(tags.join(","))
+      expect(result.length).to be 3
+      expect(result.map(&:name)).to eql tags
+      expect(result.map(&:new_record?)).to eql [false, false, false]
+      expect(result.first).to be_a Tag
+    end
+
+    it "returns old records if found and creates new record otherwise" do
+      tags = ["tag1", create(:tag).name, "tag3"]
+      result = Tag.process_tags(tags.join(","))
+      expect(result.length).to be 3
+      expect(result.map(&:name)).to eql tags
+      expect(result.map(&:new_record?)).to eql [true, false, true]
+      expect(result.first).to be_a Tag
+    end
+  end
+
+  describe "#push_representative_assignment_to_sidekiq" do
+    it "pushes parent assignment to sidekiq" do
+
+    end
+  end
+
+  describe "#push_subscription_update_to_sidekiq" do
+    context "when representative is changed" do
+      # allow(subject).to receive(:push_subscription_update_to_sidekiq)
+    end
+
+    context "when other parameters are changed" do
+
+    end
+  end
+
+  describe "#update_tag_subscriptions" do
+
+  end
   # describe ".search" do
```

Comment:
Though this was not done in this `PR`, but if it is not used, could you remove it? However, I think it will be nice to test the `.search` method, though. I think it's important.